core/services/ocr2/plugins/functions/integration_tests/v1/functions_integration_test.go

```diff
@@ -41,9 +41,43 @@ func TestIntegration_Functions_MultipleV1Requests_Success(t *testing.T) {
 	}
 
 	// config for oracle contract
-	utils.SetOracleConfig(t, owner, coordinatorContract, oracleIdentities, batchSize, &pluginConfig)
-	utils.CommitWithFinality(b)
+	utils.SetOracleConfig(t, b, owner, active.Contract, oracleIdentities, batchSize, &pluginConfig)
+
+	subscriptionId := utils.CreateAndFundSubscriptions(t, owner, linkToken, routerAddress, routerContract, clientContracts, allowListContract)
+	b.Commit()
+	utils.ClientTestRequests(t, owner, b, linkToken, routerAddress, routerContract, allowListContract, clientContracts, requestLenBytes, utils.DefaultSecretsBytes, subscriptionId, 1*time.Minute)
+}
+
+func TestIntegration_Functions_MultipleV1Requests_WithUpgrade(t *testing.T) {
+	// simulated chain with all contracts
+	owner, b, ticker, active, proposed, clientContracts, routerAddress, routerContract, linkToken, allowListContractAddress, allowListContract := utils.StartNewChainWithContracts(t, nClients)
+	defer ticker.Stop()
+
+	utils.SetupRouterRoutes(t, b, owner, routerContract, active.Address, proposed.Address, allowListContractAddress)
+
+	_, _, oracleIdentities := utils.CreateFunctionsNodes(t, owner, b, routerAddress, 39989, nOracleNodes, maxGas, nil, nil)
```

Comment:
It is odd to me there isn't a `getAvailablePort` function.

Reply:
If there was one, the port could be taken between that function call and your own bind.